packages/types/src/discord/components.ts

```diff
@@ -0,0 +1,348 @@
+/** Types for: https://discord.com/developers/docs/components/reference */
+
+import type { ChannelTypes } from './channel.js'
+
+/** https://discord.com/developers/docs/components/reference#component-object-component-types */
+export enum MessageComponentTypes {
+  /** A container for other components */
+  ActionRow = 1,
+  /** A button object */
+  Button,
+  /** A select menu for picking from choices */
+  SelectMenu,
+  /** A text input object */
+  InputText,
+  /** Select menu for users */
+  SelectMenuUsers,
+  /** Select menu for roles */
+  SelectMenuRoles,
+  /** Select menu for users and roles */
+  SelectMenuUsersAndRoles,
+  /** Select menu for channels */
+  SelectMenuChannels,
+  /** Container to display text alongside an accessory component */
+  Section,
+  /** Markdown text */
+  TextDisplay,
+  /** Small image that can be used as an accessory */
+  Thumbnail,
+  /** Display images and other media */
+  MediaGallery,
+  /** Displays an attached file */
+  File,
+  /** Component to add vertical padding between other components */
+  Separator,
+  /** Container that visually groups a set of components */
+  Container = 17,
+}
+
+export type DiscordMessageComponents = DiscordMessageComponent[]
+export type DiscordMessageComponent =
+  | DiscordActionRow
+  | DiscordSelectMenuComponent
+  | DiscordButtonComponent
+  | DiscordTextInputComponent
+  | DiscordSectionComponent
+  | DiscordTextDisplayComponent
+  | DiscordThumbnailComponent
+  | DiscordMediaGalleryComponent
+  | DiscordSeparatorComponent
+  | DiscordFileComponent
+  | DiscordContainerComponent
+
+/** https://discord.com/developers/docs/components/reference#anatomy-of-a-component */
+export interface DiscordBaseComponent {
+  /** The type of the component */
+  type: MessageComponentTypes
+  /** 32 bit integer used as an optional identifier for component */
+  id?: number
+}
+
+/** https://discord.com/developers/docs/components/reference#action-row-action-row-structure */
+export interface DiscordActionRow extends DiscordBaseComponent {
+  type: MessageComponentTypes.ActionRow
+
+  /**
+   * The components in this row
+   *
+   * @remarks
+   * Up to 5 button components, a single select component or a single text input component
+   */
+  components: (DiscordButtonComponent | DiscordSelectMenuComponent | DiscordTextInputComponent)[]
```

Comment:
Hi @Fleny113 , did you test this or something? The docs only mention up to 5 button or a single select menu component, i dont see any mention of text input component.

Reply:
My bad, i made the same mistake I told you 😅
Discord really need to fix the docs lol